src/libraries/Common/src/System/IO/Compression/ZLibNative.cs

```diff
@@ -248,8 +248,7 @@ public uint AvailOut
 
             private void EnsureNotDisposed()
             {
-                if (InitializationState == State.Disposed)
-                    throw new ObjectDisposedException(GetType().ToString());
+                ObjectDisposedException.ThrowIf(InitializationState == State.Disposed, this);
```

Comment:
The only downside of this pattern is that it no longer have a visible foldable branch for the JIT's inliner, e.g.:

```csharp
void Test(int a)
{
    if (a == 42)
        throw new Exception();
}

//
Test(43);
```

Inliner is able to detect a foldable branch here at the call-site: it sees that you pass a constant and that constant is used as a branch condition => foldable branch => we'll likely remove a lot of code if we inline `Test()` => boost inlining.

in case of
```csharp
void Test(int a)
{
    Exception.ThrowIf(a == 42);
}
```

Inliner just sees an unknown call with a random expression as its argument. However, it's smaller in terms of IL. You might say why it doesn't look into `ThrowIf` at this point - it's a good point and it's possible for AOT ("bottom-up" inlining) but it's a limitation for JIT 😞 
I am not saying we should not do it, it clearly makes code smaller and more readable, I am just explaining protentional future regressions from this (worry not! we'll catch them all during perf triage meetings 🙂) if there will be any of course.

Reply:
Thanks, but it would be very strange to throw an ODE based on a constant argument... it's almost always based on instance state. Do you see any case in this PR that could possibly be affected?

Reply:
Right, in all of this cases we use a field. However, Inliner still sometimes make assumptions that a branch against a field of a struct (I assume most cases in this PR are classes) might be foldable (or at least gives an additional boost)

Reply:
> a branch against a field of a struct (I assume most cases in this PR are classes) might be foldable

I'd be really interested in seeing a real-world case of that happening with an ODE. I'm not concerned. 

Is any of this applicable to ThrowIfNull though? I'd be more concerned if we're losing elimination of some null checks.